src/main/java/plugily/projects/buildbattle/events/MiscEvents.java

```diff
@@ -0,0 +1,64 @@
+package plugily.projects.buildbattle.events;
+
+import org.bukkit.event.Event;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.HandlerList;
+import org.bukkit.event.Listener;
+import org.bukkit.plugin.RegisteredListener;
+
+import plugily.projects.buildbattle.Main;
+import plugily.projects.minigamesbox.classic.api.event.game.PlugilyGameJoinAttemptEvent;
+import plugily.projects.minigamesbox.classic.api.event.game.PlugilyGameStopEvent;
+import plugily.projects.minigamesbox.classic.arena.ArenaState;
+
+final class MiscEvents implements Listener {
+
+	private final Main bb;
+
+	public MiscEvents(Main bb) {
+		this.bb = bb;
+		bb.getServer().getPluginManager().registerEvents(this, bb);
+	}
+
+	private boolean findThisPlugin(Event event) {
+		for (RegisteredListener registeredListener : event.getHandlers().getRegisteredListeners()) {
+			if (registeredListener.getPlugin() == bb) {
+				return true;
+			}
+		}
+
+		return false;
+	}
+
+	@EventHandler(ignoreCancelled = true)
+	public void onGameJoin(PlugilyGameJoinAttemptEvent event) {
+		if (event.getArena().getArenaState() != ArenaState.WAITING_FOR_PLAYERS) {
+			return;
+		}
+
+		if (!findThisPlugin(event)) {
+			return; // It may conflict with other Plugily Projects
+		}
+
+		for (ListenerHandler.PluginListener pluginListener : ListenerHandler.registeredListeners()) {
+			if (pluginListener.requiredForGameStart && !pluginListener.isRegistered) {
+				bb.getServer().getPluginManager().registerEvents(pluginListener.listener, bb);
+				pluginListener.isRegistered = true;
+			}
+		}
+	}
+
+	@EventHandler
+	public void onGameEnd(PlugilyGameStopEvent event) {
+		if (!findThisPlugin(event)) {
+			return; // It may conflict with other Plugily Projects
+		}
+
+		for (ListenerHandler.PluginListener pluginListener : ListenerHandler.registeredListeners()) {
+			if (pluginListener.requiredForGameStart && pluginListener.isRegistered) {
+				HandlerList.unregisterAll(pluginListener.listener);
+				pluginListener.isRegistered = false;
+			}
+		}
+	}
+}
```

Comment:
which events conflict, shouldn't we fixing the events in that case instead of activating and deactivating and still interacting with the ones. What are the isssues?

Reply:
Let's say a user has 50 plugins on 1 server including BB, and similar events are handled in these plugins, so if, for example, a plugin has a PlayerDropItemEvent, this event cancelled and it has a high priority, then this event will not be called or is called (depending on the conditions specified), only the dropped element will not be visible and the BB plugin needs this dropped item visible.

Reply:
yeah but it wouldn't change with that change, as its still a event such any which just "enables/disables". Our events are safe for our plugins and others as our only execute if on arena or other checks are made. If other plugins interact with our while it shouldn't be its a coding issue on their side.